File: SyllogismValidatorWPF/SyllogismValidatorWPF/Syllogism.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SyllogismValidatorWPF
{
    public class Syllogism
    {

        public Proposition FirstPropostion { get; set; }
        public Proposition SecondProposion { get; set; }
        public Proposition Conclusion { get; set; }
        public MajorMinorTerm majorTerm { get; set; }
        public MajorMinorTerm minorTerm { get; set; }
        public MiddleTerm middleTerm { get; set; }
        public Mood FirstMood { get; set; }
        public Mood SecondMood { get; set; }
        public Mood ConclusionMood { get; set; }
        public string Name { get; set; }
        public int Figure { get; set; }
        public bool isValid { get; set; }
        public int BrokenRule { get; set; }
        public int Id { get; set; }

        public Syllogism(int id, Mood one, Mood two, Mood concl, int figure)
        {
            Id = id;
            Figure = figure;
            FirstMood = one;
            SecondMood = two;
            ConclusionMood = concl;
            Name = string.Format(FirstMood.ToString() + SecondMood.ToString() + ConclusionMood.ToString() + "-" + Figure);
            majorTerm = new MajorMinorTerm("P");
            minorTerm = new MajorMinorTerm("S");
            middleTerm = new MiddleTerm("M");
            switch (Figure)
            {
                case 1:
                    FirstPropostion = new Proposition(one, middleTerm, majorTerm, false);
                    SecondProposion = new Proposition(two, minorTerm, middleTerm, false);
                    break;
                case 2:
                    FirstPropostion = new Proposition(one, majorTerm, middleTerm, false);
                    SecondProposion = new Proposition(two, minorTerm, middleTerm, false);
                    break;
                case 3:
                    FirstPropostion = new Proposition(one, middleTerm, majorTerm, false);
                    SecondProposion = new Proposition(two, middleTerm, minorTerm, false);
                    break;
                default:
                    FirstPropostion = new Proposition(one, majorTerm, middleTerm, false);
                    SecondProposion = new Proposition(two, middleTerm, minorTerm, false);
                    break;
            }
            Conclusion = new Proposition(concl, minorTerm, majorTerm, true);
        }

        public void CheckRules()
        {
            isValid = true;
            BrokenRule = 0;
            if ((FirstPropostion.isNegative) && (SecondProposion.isNegative))
            {
                //first rule broken
                isValid = false;
                if (BrokenRule == 0)
                {
                    BrokenRule = 1;
                }
            }
            if ((FirstPropostion.isNegative || SecondProposion.isNegative) && !Conclusion.isNegative)
            {
                //second rule broken
                isValid = false;
                if (BrokenRule == 0)
                {
                    BrokenRule = 2;
                }
            }
            if ((FirstPropostion.isUniversal && SecondProposion.isUniversal) && !Conclusion.isUniversal)
            {
                //third rule broken
                isValid = false;
                if (BrokenRule == 0)
                {
                    BrokenRule = 3;
                }
            }
            if (!middleTerm.isDistributed)
            {
                //fourth rule broken
                isValid = false;
                if (BrokenRule == 0)
                {
                    BrokenRule = 4;
                }
            }
            if ((!(majorTerm as MajorMinorTerm).isDistributed && (majorTerm as MajorMinorTerm).isDistributedInConclusion) || (!(minorTerm as MajorMinorTerm).isDistributed && (minorTerm as MajorMinorTerm).isDistributedInConclusion))
            {
                //fifth rule broken
                isValid = false;
                if (BrokenRule == 0)
                {
                    BrokenRule = 5;
                }
            }
        }
        public string Write()
        {
            return string.Format("\n" + FirstPropostion.Write() + "\n" + SecondProposion.Write() + "\n" + Conclusion.Write() + "\n\n");
        }
    }
}

File: app/models/syllogism.ts
import { Proposition, Mood } from "validator/models/proposition";
import { MajorMinorTerm, MiddleTerm } from "validator/models/term";

export class Syllogism {
    public firstProposition: Proposition;
    public secondProposition: Proposition;
    public conclusion: Proposition;
    public majorTerm: MajorMinorTerm
    public minorTerm: MajorMinorTerm;
    public middleTerm: MiddleTerm;
    public name: string;
    public isValid: boolean;
    public brokenRule: number;
    public description: string;
    public cssClass: string;

    constructor(public id: number, public firstMood: Mood, public secondMood: Mood, public conclusionMood: Mood, public figure: number) {
        this.name = this.getName();
        this.majorTerm = new MajorMinorTerm("P");
        this.minorTerm = new MajorMinorTerm("S");
        this.middleTerm = new MiddleTerm("M");
        switch (figure) {
            case 1:
                this.firstProposition = new Proposition(firstMood, this.middleTerm, this.majorTerm, false);
                this.secondProposition = new Proposition(secondMood, this.minorTerm, this.middleTerm, false);
                break;
            case 2:
                this.firstProposition = new Proposition(firstMood, this.majorTerm, this.middleTerm, false);
                this.secondProposition = new Proposition(secondMood, this.minorTerm, this.middleTerm, false);
                break;
            case 3:
                this.firstProposition = new Proposition(firstMood, this.middleTerm, this.majorTerm, false);
                this.secondProposition = new Proposition(secondMood, this.middleTerm, this.minorTerm, false);
                break;
            default:
                this.firstProposition = new Proposition(firstMood, this.majorTerm, this.middleTerm, false);
                this.secondProposition = new Proposition(secondMood, this.middleTerm, this.minorTerm, false);
                break;
        }
        this.conclusion = new Proposition(conclusionMood, this.minorTerm, this.majorTerm, true);
        this.brokenRule = this.getBrokenRule();
        this.isValid = !!this.brokenRule;
        this.description = this.write();
        this.cssClass = `${this.firstMood}${this.secondMood}${this.conclusionMood}${this.figure}`;
    }

    public getName(): string {
        return this.firstMood.toString() + this.secondMood.toString() + this.conclusionMood.toString() + "-" + this.figure;
    }

    public getBrokenRule(): number {
        if ((this.firstProposition.isNegative) && (this.secondProposition.isNegative)) {
            //first rule broken
            return 1;
        }
        if ((this.firstProposition.isNegative || this.secondProposition.isNegative) && !this.conclusion.isNegative) {
            //second rule broken
            return 2;
        }
        if ((this.firstProposition.isUniversal && this.secondProposition.isUniversal) && !this.conclusion.isUniversal) {
            //third rule broken
            return 3;
        }
        if (!this.middleTerm.isDistributed) {
            //fourth rule broken
            return 4;
        }
        if ((!(this.majorTerm as MajorMinorTerm).isDistributed && (this.majorTerm as MajorMinorTerm).isDistributedInConclusion) || (!(this.minorTerm as MajorMinorTerm).isDistributed && (this.minorTerm as MajorMinorTerm).isDistributedInConclusion)) {
            //fifth rule broken
            return 5;
        }
        return 0;
    }
    public write(): string
    {
        return `
        ${this.firstProposition.description}
        ${this.secondProposition.description}
        ${this.conclusion.description}

        `;
    }
}